packages/vscode-container-client/src/clients/DockerClientBase/parsePruneLikeOutput.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { tryParseSize } from "./tryParseSize";
+
+// default regex to parse the space reclaimed
+const PruneSpaceReclaimedRegex = /Total space reclaimed: ([\w\s]+)/i;
+const LineRegex = /^\w+$/igm;
+
+export type PruneParseOptions = {
+    // if space reclaimed regex is not provided, we'll use the default
+    spaceReclaimedRegex?: RegExp;
+    // if resource regex is not provided, we'll assume the resource is the line itself
+    resourceRegex?: RegExp;
+};
+
+export type PruneResult = {
+    spaceReclaimed: number;
+    resources: string[];
+};
+
+export function parsePruneLikeOutput(output: string, options: PruneParseOptions): PruneResult {
+    const resourceRegex = options?.resourceRegex || LineRegex;
+    const spaceReclaimedRegex = options?.spaceReclaimedRegex || PruneSpaceReclaimedRegex;
+
+    let deletedResources: string[] = [];
+    let spaceReclaimed: number = 0;
+
+    // match resources
+    deletedResources = output.match(resourceRegex) || [];
+
+    // remove the space reclaimed line if it exists
+    if (deletedResources.length > 0) {
+        deletedResources.pop();
+    }
```

Comment:
You should craft the regexp so this step isn't needed. i.e. `LineRegex` or `resourceRegex` should not match `"Total space reclaimed: 1234 MB"`

Reply:
you're also right here as well. through some more testing, I realized this step is actually messes things up, going to remove iti.